lib/mobility-core/src/Kernel/External/AadhaarVerification/Interface/Gridline.hs

```diff
@@ -0,0 +1,126 @@
+import Kernel.External.AadhaarVerification.Gridline.Config
+import qualified Kernel.External.AadhaarVerification.Gridline.Types as GT
+import Kernel.External.AadhaarVerification.Interface.Types
+import Kernel.External.Encryption
+import Kernel.Prelude
+import Kernel.Tools.Metrics.CoreMetrics (CoreMetrics)
+
+generateAadhaarOtp ::
+  ( EncFlow m r,
+    CoreMetrics m
+  ) =>
+  GridlineCfg ->
+  AadhaarOtpReq ->
+  m AadhaarVerificationResp
+generateAadhaarOtp cfg req = do
+  let url = cfg.url
+  apiKey <- decrypt cfg.apiKey
+  authType <- decrypt cfg.authType
+  let reqData =
+        GT.GridlineAadhaarOtpReq
+          { aadhaar_number = req.aadhaarNumber,
+            consent = req.consent
+          }
+  gridLineVerificationSuccess <- Gridline.generateAadhaarOtp url apiKey authType reqData
+  case (gridLineVerificationSuccess._data, gridLineVerificationSuccess._error) of
+    (Just _data, Nothing) -> do
+      pure $
+        AadhaarVerificationResp
+          { statusCode = _data.code,
+            message = _data.message,
+            transactionId = _data.transaction_id,
+            requestId = gridLineVerificationSuccess.request_id
+          }
+    (_, Just _error) -> do
+      pure $
+        AadhaarVerificationResp
+          { statusCode = _error.code,
+            message = _error.message,
+            transactionId = " ",
+            requestId = gridLineVerificationSuccess.request_id
+          }
+    (_, _) -> do
+      pure $
+        AadhaarVerificationResp
+          { statusCode = " ",
+            message = "error occurred in the interface side",
+            transactionId = " ",
+            requestId = " "
+          }
+
+verifyAadhaarOtp ::
+  ( EncFlow m r,
+    CoreMetrics m
+  ) =>
+  GridlineCfg ->
+  AadhaarOtpVerifyReq ->
+  m AadhaarOtpVerifyRes
+verifyAadhaarOtp cfg req = do
+  let url = cfg.url
+  apiKey <- decrypt cfg.apiKey
+  authType <- decrypt cfg.authType
+  let transactionId = req.transactionId
+  let reqData =
+        GT.GridlineAadhaarOtpVerifyReq
+          { otp = req.otp,
+            include_xml = req.includeXml,
+            share_code = req.shareCode
+          }
+  gridLineVerificationSuccess <- Gridline.verifyAadhaarOtp url apiKey authType transactionId reqData
+  case (gridLineVerificationSuccess._data, gridLineVerificationSuccess._error) of
+    (Just _data, Nothing) -> do
+      pure $
+        AadhaarOtpVerifyRes
+          { transactionId = _data.transaction_id,
+            message = _data.message,
+            code = _data.code,
+            name = _data.aadhaar_data.name,
+            gender = _data.aadhaar_data.gender,
+            date_of_birth = _data.aadhaar_data.date_of_birth,
+            share_code = _data.share_code,
+            image = _data.aadhaar_data.photo_base64,
+            request_id = gridLineVerificationSuccess.request_id
+          }
+    (_, Just _error) -> do
+      pure $
+        AadhaarOtpVerifyRes
```

Comment:
why like this?

Reply:
because in the response the error and data will toggle there    code and message   is present inside therefore like that

Reply:
same.